File: Global.fs
namespace BloodFeverFs

open KSGodot.Audio

open Godot
open KSGodotUtils

type Global() =
    inherit Node()

    member val LastSceneSnapshot: Option<ImageTexture> = None with get, set
    member val LastSceneLayer: Option<CanvasLayer> = None with get, set
    member val LastScene: Option<Node2D> = None with get, set


    member this.BackgroundAudio() =
        this.getNode<BackgroundAudio> ("/root/Audio")

File: Global.ts
import BackgroundAudio from "./ksgodot/audio/BackgroundAudio";

export default class Global extends Node {
  lastSceneSnapshot: ImageTexture | null = null;
  lastSceneLayer: CanvasLayer | null = null;
  lastScene: Node2D | null = null;

  get BackgroundAudio() {
    return this.get_node<BackgroundAudio>("/root/AudioManager");
  }
}
